### DISCUSSION TITLE
Simple filter to run a text string through an R function

### DISCUSSION THREAD

**Discussion by harrelfe** [quarto-dev/quarto-cli#9431] [Q&A] [answered]
### Description

I would like to have a simple filter that allows me to process text strings with a user-specified function.  For example I'd like

```
This is {=xx a string of text that is arbitrarily long but is one line}.
```

to result in 

```
`r xx("a string of text that is arbitrarily long but is one line")`.
```

Is this possible and can anyone provide a simple how-to?  The = delimiter before the function name can be anything that doesn't conflict with existing filters.

**Comment by mcanouil** [comment]
Could you explain a bit more your use case and/or your goal with such filter?
I don't understand what this adds to the inline R call, i.e., `` `{r} f(arg)` `` (it's not shorter, nor simpler and it would hide the computing language used).

It feels more complicated and I am not even sure you can do that because, this would require the computation to happen after this filter.

**Reply by harrelfe** [reply]
Thanks for responding.  I'd like to use simple functions to process a number of simple markups, with R handling logic for the output format of html vs docx.  The goal is to not use all the quote marks needed by a regular inline call, e.g., the call is not as simple as you wrote but is instead `r f("arg")`.  The double quotes get in the way of the visual flow, and it would be good to not have to use backticks.  I know this is minor cosmetics but I'd like to use this if it's not hard to implement.  I can do a lot of markups with divs but I'm needing multiple different markups within a paragraph.

Occasionally I'll also use a function to send output to a different destination than the current report flow.

**Reply by mcanouil** [reply, marked as answer]
See:
- https://pandoc.org/lua-filters.html
- https://quarto.org/docs/extensions/lua.html
- https://quarto.org/docs/prerelease/1.4/lua_changes.html#more-precise-targeting-of-ast-processing-phases
- https://quarto.org/docs/extensions/lua-api.html

If feels way more complicated that it needs to be and only for a "quotes issue".
See below an example, showing that it won't work as the earlier is pre AST not pre-computations (I might be missing something, but don't think so).

```qmd
---
title: "Quarto Playground"
format: html
filters:
  - at: pre-ast
    path: filter.lua
engine: knitr
---

This is a playground for Quarto.

This is `a string of text that is arbitrarily long but is one line`{=xx}.

`{r} 1+1`
```

```lua
function RawInline(raw)
  if raw.format == "xx" then
    return pandoc.Code("{r} identity('" .. raw.text .. "')")
  end
end
```

**Reply by harrelfe** [reply]
Thanks very much!  I defer to your expertise.  I didn't realize the computation order issue.

**Reply by cscheid** [reply]
Yes, Lua filters run _after_ the execution engines. We have (long-term) plans to add a pre-engine Pandoc pass, but that will not happen for at least a year.
